scripts/jira/start_ticket.py

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env python
+
+"""Tool for bringing a ticket into the current sprint and updating its status.
+
+Ticket ID
+---------
+
+The default ticket ID discovered by parsing the current branch name,
+looking for the pattern
+
+  <project-id>-<ticket-num>[-<description>]
+
+For example
+
+  USHIFT-1069-jira-manage-ticket
+
+produces a ticket ID of
+
+  USHIFT-1069
+
+Set a ticket ID explicitly using the `--ticket-id` option.
+
+Sprint
+------
+
+The ticket is added to the current active sprint by looking at the
+sprints visible through the MicroShift Scrum Board
+(https://issues.redhat.com/secure/RapidBoard.jspa?rapidView=14885) and
+finding the active sprint beginning with "uShift". Disable this
+behavior using the `--no-sprint` flag.
+
+Target Version
+--------------
+
+The `--target-version` flag can be used to set a target version if one
+is not set already.
+
+Status
+------
+
+Use the `--in-progress` or `--review` options to control the
+status. The default is `--review`.
+
+Authentication
+--------------
+
+The script requires a Jira token passed via the `JIRA_API_TOKEN`
+environment variable. See README.md for details of creating the token.
+
+"""
+
+import argparse
+import os
+import subprocess
+
+import jira
+
+SERVER_URL = 'https://issues.redhat.com/'
+SCRUM_BOARD = 14885
+
+
+def custom_field_manager(server):
+    """Return callables for working with custom fields.
+
+    Custom fields are stored on the ticket in ticket.fields with names
+    like "customfield_12319940". The information about those fields
+    can be queried from the Jira API to map the user-visible names
+    (like "Target Version") to the less helpful custom field attribute
+    name. This function looks up the mappings and then creates two
+    closures that can get and set values from a ticket using the
+    understandable names.
+
+    The getter takes a ticket and custom field name and returns the
+    value of the field for that ticket.
+
+    The setter takes a ticket, custom field name, and new value and
+    updates the ticket to set that field to the new value.
+
+    """
+    field_info = server.fields()
+    fields_by_name = {
+        f['name']: f
+        for f in field_info
+    }
+    def get_field_value(ticket, name):
+        field_details = fields_by_name[name]
+        return getattr(ticket.fields, field_details['id'])
+    def set_field_value(ticket, name, value):
+        field_details = fields_by_name[name]
+        return ticket.update(fields={
+            field_details['id']: value,
+        })
+    return get_field_value, set_field_value
+
+
+def get_active_sprint(server):
+    """Return the active sprint for the USHIFT project."""
+    valid_sprints = server.sprints(SCRUM_BOARD)
```

Comment:
You can include `state = active` and that should save the filtering below. There is a maxResults that defaults to 50, which for us means 150 weeks, so it will stop working next year?

Reply:
Does the state filter fix that or are you saying I need to ask for more results?

Reply:
The USHIFT board includes sprints from multiple projects, so we still need the loop. I guess I can remove the date checking portion, so I've done that.